meta-ostro/lib/supportedrecipes/supportedrecipes.py

```diff
@@ -0,0 +1,346 @@
+
+def check_build(d, event):
+    supported_recipes, files = load_supported_recipes(d)
+    supported_recipes_check = d.getVar('SUPPORTED_RECIPES_CHECK', True)
+    if not supported_recipes_check:
+        return
+
+    # Always add a trailing $ to ensure a full match.
+    isnative_exception = re.compile('(' + '|'.join(d.getVar('SUPPORTED_RECIPES_NATIVE_RECIPES', True).split()) + ')$')
+    isnative_baseclasses = d.getVar('SUPPORTED_RECIPES_NATIVE_BASECLASSES', True).split()
+    valid = ('note', 'warn', 'error', 'fatal')
+    if supported_recipes_check not in valid:
+        bb.fatal('SUPPORTED_RECIPES_CHECK must be set to one of %s, currently is: %s' % ('/'.join(valid), supported_recipes_check))
+    logger = bb.__dict__[supported_recipes_check]
+
+    # See bitbake/lib/bb/cooker.py buildDependTree() for the content of the depgraph hash.
+    # Basically it mirrors the information dumped by "bitbake -g".
+    depgraph = event._depgraph
+    # import pprint
+    # bb.note('depgraph: %s' % pprint.pformat(depgraph))
+
+    dirname = d.getVar('SUPPORTED_RECIPES_SOURCES_DIR', True)
+    report_sources = d.getVar('SUPPORTED_RECIPES_SOURCES', True)
+
+    unsupported = {}
+    sources = []
+    for pn, pndata in depgraph['pn'].iteritems():
+        # We only care about recipes compiled for the target.
+        # Most native ones can be detected reliably because they inherit native.bbclass,
+        # but some special cases have to be hard-coded.
+        # Image recipes also do not matter.
+        def isnative():
+            for inherited in pndata['inherits']:
+                if os.path.basename(inherited) in isnative_baseclasses:
+                    return True
+            # Some build recipes do not inherit cross.bbclass and must be skipped explicitly.
+            # The "real" recipes (in cases like glibc) still get checked. Other recipes are OE-core
+            # internal helpers.
+            if isnative_exception.match(pn):
+                return True
+        if not isnative():
+            filename = pndata['filename']
+            collection = bb.utils.get_file_layer(filename, d)
+            supported = supported_recipes.recipe_supported(pn, collection)
+            if not supported:
+                unsupported[pn] = collection
+            if report_sources:
+                dumpfile = dirname + '/' + pn + filename
+                with open(dumpfile) as f:
+                    reader = csv.reader(f)
+                    for row in reader:
+                        row_hash = dict([(f, row[i]) for i, f in enumerate(SOURCE_FIELDS)])
+                        row_hash['supported'] = 'yes' if supported else 'no'
+                        sources.append(row_hash)
+
+    if report_sources:
+        with open(report_sources, 'w') as f:
+            fields = SOURCE_FIELDS
+            # Insert after 'collection'.
+            fields.insert(fields.index('collection') + 1, 'supported')
```

Comment:
this would modify global constant instead of local copy "fields".
```python
In [1]: SOURCE_FIELDS="a,b,c,d".split(",")
In [2]: f=SOURCE_FIELDS
In [3]: f.insert(f.index("c")+1, "something")
In [4]: SOURCE_FIELDS
Out[4]: ['a', 'b', 'c', 'something', 'd']
```